src/coreclr/vm/excep.cpp

```diff
@@ -6533,59 +6533,48 @@ AdjustContextForJITHelpers(
 
 #if defined(USE_FEF) && !defined(TARGET_UNIX)
 
-struct SavedExceptionInfo
+static void FixContextForFaultingExceptionFrame(
+    EXCEPTION_POINTERS* ep,
+    EXCEPTION_RECORD* pOriginalExceptionRecord,
+    CONTEXT* pOriginalExceptionContext)
 {
-    EXCEPTION_RECORD m_ExceptionRecord;
-    CONTEXT m_ExceptionContext;
-    CrstStatic m_Crst;
-
-    void SaveExceptionRecord(EXCEPTION_RECORD *pExceptionRecord)
-    {
-        LIMITED_METHOD_CONTRACT;
-        size_t erSize = offsetof(EXCEPTION_RECORD, ExceptionInformation) +
-            pExceptionRecord->NumberParameters * sizeof(pExceptionRecord->ExceptionInformation[0]);
-        memcpy(&m_ExceptionRecord, pExceptionRecord, erSize);
-
-    }
+    WRAPPER_NO_CONTRACT;
 
-    void SaveContext(CONTEXT *pContext)
-    {
-        LIMITED_METHOD_CONTRACT;
-#ifdef CONTEXT_EXTENDED_REGISTERS
+    // don't copy param args as have already supplied them on the throw
```

Comment:
This is copying `NumberParameters` that is not strictly necessary according to this comment.

I think it would look better to just explicitly assign the 4 fields that this needs to copy.

Reply:
Here I just restored the code that was deleted by #70428 and that is known to work.  If we are seeing the exception we just raised, I think we actually need to overwrite just one field, `ExceptionAddress`.  All other fields should have the values we passed to `RaiseException`.